LanguageFeatures/Patterns/type_inference_A16_t02.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Calculate the static type of the pattern. Using that value type,
+/// recurse through the pattern again downwards to the leaf subpatterns filling
+/// in any missing types in the pattern. This process may also insert implicit
+/// coercions and casts from dynamic when values flow into a pattern during
+/// matching.
+///
+/// @description Check that the calculation of the static type of a
+/// parenthesized pattern performs implicit coercions
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns
+
+import "../../Utils/expect.dart";
+import "../../Utils/static_type_helper.dart";
+
+main() {
+  var (<double>[v1]) = [42];
```

Comment:
I think this is not a phase 3 test: The type schema of the pattern is `List<double>`, which will be used during inference of the initializing expression, and this causes '42' to be coerced to a `double`. All those things are testing phase 1 and 2, not phase 3.

So the only phase 3 thing which is tested here is the declared type of `v1`.

Reply:
OK!